packages/backend/src/modules/tvl2/api/Tvl2Controller.ts

```diff
@@ -682,51 +683,110 @@ export class Tvl2Controller {
   async getTokenChart(
     token: { address: EthereumAddress | 'native'; chain: string },
     project: ProjectId,
-  ) {
+    lastHour: UnixTime,
+  ): Promise<TokenTvlApiCharts> {
     const projectAmounts = this.amountConfig.get(project)
-    if (!projectAmounts) return
+    assert(projectAmounts)
 
     const assetId = createAssetId(token)
     const amountConfigs = projectAmounts.filter(
       (x) => createAssetId(x) === assetId,
     )
 
-    const amounts = await this.amountRepository.getDailyByConfigId(
+    const projectConfig = this.projects.find((x) => x.id === project)
+    assert(projectConfig, 'Project not found!')
+
+    const amounts = await this.amountRepository.getByConfigIdsInRange(
       amountConfigs.map((x) => x.configId),
+      projectConfig.minTimestamp,
+      lastHour,
     )
     const priceConfig = this.priceConfigs.get(assetId)
     assert(priceConfig, 'PriceId not found!')
-    const prices = await this.priceRepository.getDailyByConfigId(
-      priceConfig.priceId,
+    const prices = await this.priceRepository.getByConfigIdsInRange(
+      [priceConfig.priceId],
+      projectConfig.minTimestamp,
+      lastHour,
     )
     const pricesMap = new Map(
       prices.map((x) => [x.timestamp.toNumber(), x.priceUsd]),
     )
-
     const amountsByTimestamp = groupBy(amounts, 'timestamp')
 
-    const values: [number, number][] = []
+    const values = new Map<number, { amount: number; value: number }>()
     for (const [timestamp, amounts] of Object.entries(amountsByTimestamp)) {
```

Comment:
Here instead of iterating over entries iterate over timestamps (use getTimestampsForProject)